python/search_completer/hierarchical_search_completer.py

```diff
@@ -0,0 +1,187 @@
+# Copyright (c) 2017 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+import sgtk
+from sgtk.platform.qt import QtCore, QtGui
+
+from .search_completer import SearchCompleter
+
+shotgun_model = sgtk.platform.import_framework("tk-framework-shotgunutils", "shotgun_model")
+
+
+class HierarchicalSearchCompleter(SearchCompleter):
+    """
+    A standalone :class:`PySide.QtGui.QCompleter` class for matching SG entities to typed text.
+
+    :signal: ``node_activated(str, int, str, str, list)`` - Fires when someone activates a
+        node inside the search results. The parameters are ``type``, ``id``, ``name``,
+        ``label path`` and ``incremental_paths``. If the node activated is not an entity,
```

Comment:
I think having a mode would be better. Its simple to filter out entries, plus we can't tell what people would like to do with the widget. Maybe I can add this as a separate PR to simplify the review process?